src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,45 +1,109 @@
-import { useState } from 'react';
+import React, { useState } from 'react';
 import { TextField } from '../TextField';
+import { Movie } from '../../types/Movie';
 
-export const NewMovie = () => {
-  // Increase the count after successful form submission
-  // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+interface Props {
+  onAdd: (movie: Movie) => void;
+}
+
+const initialFormState = {
+  title: '',
+  description: '',
+  imgUrl: '',
+  imdbUrl: '',
+  imdbId: '',
+};
+
+export const NewMovie: React.FC<Props> = ({ onAdd }) => {
+  const [count, setCount] = useState(0);
+
+  const [formState, setFormState] = useState(initialFormState);
+
+  const handleInputChange = (
+    value: string,
+    name: string,
+  ) => {
+    setFormState(state => ({
+      ...state,
+      [name]: value,
+    }));
+  };
+
+  const resetForm = () => {
+    setFormState(initialFormState);
+  };
+
+  const {
+    title,
+    description,
+    imgUrl,
+    imdbUrl,
+    imdbId,
+  } = formState;
+
+  const addNewMovie = () => {
+    const movie = {
+      title,
+      description,
+      imgUrl,
+      imdbUrl,
+      imdbId,
+    };
+
+    onAdd(movie);
+    resetForm();
+
+    setCount((prevCount) => prevCount + 1);
+  };
 
   return (
-    <form className="NewMovie" key={count}>
+    <form
+      className="NewMovie"
+      key={count}
+      onSubmit={(event) => {
+        event.preventDefault();
+        addNewMovie();
+      }}
```

Comment:
Let's better create a handler here and for better readability leave just 
```suggestion
      onSubmit={handleSubmit}
```

Reply:
fixed